Source/Tests/LinuxSyscalls/Syscalls.h

```diff
@@ -349,6 +349,18 @@ uint64_t CloneHandler(FEXCore::Core::CpuStateFrame *Frame, FEX::HLE::clone3_args
     return flags;
   }
 
+
+template<typename T>
+static const bool HasSyscallError(T Result) {
+  // Most raw syscalls return errors using the negative range
+  // These are guaranteed to not return a valid result in the range of [-4095, -1]
+  // glibc typically wraps these results and sets errno to -Result and returns -1 from the handler
+  // In cases where FEX needs to use raw syscalls, this helper checks for this idiom
+  // !!! WARNING !!!
+  // Some syscalls may return valid results in this range, like ioctl
+  // Don't use this in those cases
+  return reinterpret_cast<uint64_t>(Result) > -4096;
```

Comment:
Why is `reinterpret_cast` needed here over `static_cast`?

Reply:
Because we also pass void* in through this interface with mmap routines.

Reply:
Are we? All the uses below appear to cast the result to `uint64_t` before call site.

If I missed one that doesn't, would the following work?
```
static bool HasSyscallError(uint64_t Result) {
    return Result > 0xff..0000;
}
static bool HasSyscallError(void* Result) {
    return HasSyscallError(static_cast<uintptr_t>(Result));
}
```

That would also yield a more constrained interface, where accidentally passing the wrong argument type won't make the compiler explode.

Reply:
`Source/Tests/LinuxSyscalls/LinuxAllocator.cpp:390:17` uses it as a `void*` Yes.

Reply:
Changed to fit constrained used case.